torch/_subclasses/meta_utils.py

```diff
@@ -280,10 +295,13 @@ def __call__(self, t, shape_env=None):
                 # tests all break so we just exclude this.  In any case
                 # the to conversion isn't really right anyhow.
                 self.miss += 1
+                if strict:
+                    return NotImplemented
                 return t
             else:
                 self.hit += 1
-                r = self.meta_tensor(t, shape_env=shape_env)
+                r = self.meta_tensor(t, shape_env=shape_env, callback=callback)
+                # TODO: this is suspicious, now that we have callback argument
                 if type(t) is torch.nn.Parameter:
                     r = torch.nn.Parameter(r, requires_grad=r.requires_grad)
                 return r
```

Comment:
Should we move this logic https://github.com/pytorch/pytorch/blob/48e1981d0286c75afe043750c3f3b4f6b841405c/torch/_subclasses/fake_tensor.py#L235-L239 here ?

Reply:
Oh yeah, good catch

Reply:
I moved it into meta_tensor itself, instead of at top level. This way recursive grad field also works.